perfmetrics/scripts/generate_files.py

```diff
@@ -7,29 +7,49 @@
 import argparse
 import configparser
 from datetime import datetime as dt
+import logging
 import os
 import subprocess
 from subprocess import Popen
 import sys
 
 OUTPUT_FILE = str(dt.now().isoformat()) + '.out'
-TEMPORARY_DIRECTORY = '/tmp/data_gen'
+TEMPORARY_DIRECTORY = './tmp/data_gen'
 BATCH_SIZE = 100
 
 
-def logmessage(message) -> None:
-  with open(OUTPUT_FILE, 'a') as out:
-    out.write(message)
-  print(message)
+logging.basicConfig(
+    level=logging.INFO,
+    format='%(asctime)s [%(levelname)s] %(message)s',
+    handlers=[logging.StreamHandler(sys.stdout)],
+)
+logger = logging.getLogger()
+
+
+def logmessage(message, output_format) -> None:
+  if output_format == 'output_file':
+    with open(OUTPUT_FILE, 'a') as out:
+      out.write(message)
+    print(message)
+  elif output_format == 'return':
+    logger.info(message)
 
 
 def generate_files_and_upload_to_gcs_bucket(destination_blob_name, num_of_files,
                                             file_size_unit, file_size,
                                             filename_prefix,
-                                            local_destination_folder):
-
-  for batch_start in range(1, num_of_files+1, BATCH_SIZE):
+                                            local_destination_folder,
+                                            upload_to_gcs_bucket,
+                                            logging_method):
+
+  # Iterating till num_files will result in some files not being created.
+  # So iterating till num_of_files + BATCH_SIZE + 1 as a safe measure to cover
+  # all files. Manually breaking the loop at correct time using if statements inside.
+  for batch_start in range(1, num_of_files + BATCH_SIZE + 1, BATCH_SIZE):
```

Comment:
can you explain in what scenario it will break

Reply:
changed to the earlier loop with if statement to break, inside the inner for loop.